src/main/java/org/jenkinsci/plugins/github_branch_source/AbstractGitHubSCMSource.java

```diff
@@ -336,27 +344,40 @@ public ListBoxModel doFillRepositoryItems(@AncestorInPath SCMSourceOwner context
                     }
                     return result;
                 }
+
+                GHOrganization org = null;
+                try {
+                    org = github.getOrganization(repoOwner);
+                } catch (FileNotFoundException fnf) {
+                    LOGGER.log(Level.FINE, "There is not any GH Organization named " + repoOwner);
```

Comment:
@jglick I thought to improve the Form validation in a different PR. I'll include this case.